Loop/Managers/WatchDataManager.swift

```diff
@@ -40,6 +43,52 @@ final class WatchDataManager: NSObject {
     }()
 
     private var lastSentSettings: LoopSettings?
+    
+    let healthStore: HKHealthStore
+    let sleepStore: SleepStore
+    
+    var lastBedtimeQuery: Date {
+        didSet {
+            UserDefaults.appGroup?.lastBedtimeQuery = lastBedtimeQuery
+        }
+    }
+    
+    var bedtime: Date? {
+        didSet {
+            UserDefaults.appGroup?.bedtime = bedtime
+        }
+    }
+    
+    private func updateBedtimeIfNeeded() {
+        let now = Date()
+        let lastUpdateInterval = now.timeIntervalSince(lastBedtimeQuery)
+        let calendar = Calendar.current
+        
+        guard lastUpdateInterval >= TimeInterval(hours: 24) else {
+            // increment the bedtime by 1 day if it's before the current time, but we don't need to make another HealthKit query yet
+            if let bedtime = bedtime, bedtime < now {
+                let hourComponent = calendar.component(.hour, from: bedtime)
+                let minuteComponent = calendar.component(.minute, from: bedtime)
+                
+                if let newBedtime = calendar.nextDate(after: now, matching: DateComponents(hour: hourComponent, minute: minuteComponent), matchingPolicy: .nextTime) {
```

Comment:
Here's another case where `newBedtime` could be more than a day in the future in case of DST. We could add a `newBedtime.timeIntervalSince(bedtime) <= .hours(24)` clause similarly to before; alternatively, see my comment in `complicationUserInfoTransferInterval` below.